eventually/topic/views.py

```diff
@@ -5,12 +5,103 @@
+                                  RESPONSE_400_INVALID_DATA,
+                                  RESPONSE_404_OBJECT_NOT_FOUND)
+from .models import Topic
 
 
 class TopicView(View):
     """
     Item view that handles GET, POST, PUT, DELETE requests and provides appropriate
     operations with topic model.
     """
+
+    def get(self, request, topic_id=None, curriculum_id=None,):
+        """
+        Method that handles GET request.
+
+        :param request: the accepted HTTP request.
+        :type request: `HttpRequest object`
+
+        :param topic_id: ID of the certain topic.
+        :type topic_id: `int`
+
+        :param curriculum_id: ID of the certain curriculum.
+        :type curriculum_id: `int`
+
+        :return: the response with the certain topic information when topic_id was transferred or
+                 the full list of certain curriculum topics. If topic_id or curriculum_id does
+                 not exist returns the 404 failed status code response.
+        :rtype: `HttpResponse object.
+        """
+
+        if topic_id:
+            topic = Topic.get_by_id(topic_id)
+            if not topic:
+                return RESPONSE_404_OBJECT_NOT_FOUND
+
+            topic = topic.to_dict()
+            return JsonResponse(topic, status=200)
+
+        if curriculum_id:
+            curriculum = Curriculum.get_by_id(curriculum_id)
+            if not curriculum:
+                return RESPONSE_404_OBJECT_NOT_FOUND
+            topics = curriculum.topic_set.all()  # Topic.object.filter(curriculum=curriculum_id)
+            data = {'topics': [topic.to_dict() for topic in topics]}
+            return JsonResponse(data, status=200)
+
+    def post(self, request, curriculum_id):
+        """
+        Method that handles POST request.
+
+        :param request: the accepted HTTP request.
+        :type request: `HttpRequest object`
+
+        :param curriculum_id: ID of the certain curriculum.
+        :type curriculum_id: `int`
+
+        :return: the response with certain topic information when the topic was successfully
+                 created or response with 400 or 404 failed status code.
+        :rtype: `HttpResponse object."""
+
+        author = request.user
+        data = request.body
+        curriculum = Curriculum.get_by_id(curriculum_id)
+
+        if not data:
+            return RESPONSE_400_INVALID_DATA
+
+        if not curriculum:
+            return RESPONSE_404_OBJECT_NOT_FOUND
+
+        data = {'title': data.get('title'),
+                'description': data.get('description') if data.get('description') else '',
+                'mentors': data.get('mentors') if data.get('mentors') else []}
+
+        topic = Topic.create(curriculum, author, **data)
+        if topic:
+            return JsonResponse(topic.to_dict(), status=201)
+
+        return RESPONSE_400_DB_OPERATION_FAILED
+
+    def put(self, request, curriculum_id=None, topic_id=None):
```

Comment:
return http status` 501 Not Implemented`

Reply:
We haven't such response in responsehelper.py. Do I have to add this response?